### DISCUSSION TITLE
Traffic management

### DISCUSSION THREAD

**Discussion by HardyBenjamin** [openTCS/opentcs#101] [Q&A]
Hello everyone,

I'm curious about the traffic management capabilities of the OpenTCS software. Could someone provide insights into how OpenTCS handles traffic management for AGVs in complex environments? Specifically, I'm interested in understanding how the software avoids deadlocks, manages AGV movements across nodes and edges, and optimizes routes to ensure efficient and collision-free transportation.

Thank you in advance for sharing your knowledge and experiences!

**Comment by swltr** [comment]
@HardyBenjamin The [User's Guide](https://opentcs.org/docs/5.9/user/opentcs-users-guide.html) and the [Developer's Guide](https://opentcs.org/docs/5.9/developer/developers-guide/opentcs-developers-guide.html) contain a lot of information. If any important information is missing there, specific suggestions for improvement of that documentation (and PRs for making it happen) are always welcome, of course.
